### PR TITLE
Simplify build image Makefile's

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
We now create a builder in any case for builds on non x86_64 machines.
We can also avoid temp directories by not copying content around.
#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by cpanato — APPROVED** [review]
/lgtm
/hold for more eyes

File: images/build/debian-iptables/Makefile [hunk 1]
```diff
@@ -39,14 +38,13 @@ export DOCKER_CLI_EXPERIMENTAL := enabled
 SUDO=$(if $(filter 0,$(shell id -u)),,sudo)
 
 build:
-	cp -r ./$(CONFIG) $(TEMP_DIR)/
+	# Fix possible issues with the local umask
+	umask 0022
 
-ifneq ($(ARCH),amd64)
```
Comment (hasheddan) [inline images/build/debian-iptables/Makefile:44]: What was are reasoning for not doing this on `amd64` in the past?

Comment (saschagrunert) [reply images/build/debian-iptables/Makefile:44]: We assumed that this was the native architecture, which is not necessarily always the case.

Comment (hasheddan) [reply images/build/debian-iptables/Makefile:44]: Gotcha 👍

**Review by hasheddan — APPROVED** [review]
/lgtm

**Comment by justaugustus** [comment]
Nice cleanup!
/lgtm
/approve
/hold cancel

**Comment by justaugustus** [comment]
@saschagrunert --

From https://kubernetes.slack.com/archives/C2C40FMNF/p1613430854267700?thread_ts=1613419589.265800&cid=C2C40FMNF:

> The go1.16rc1 ([update](https://github.com/kubernetes/release/pull/1917)) hit our CI container and the go1.15 and go1.16 go.sums differ, so hack/verify-go-mod.sh bombed out (in https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/release/1914/pull-release-verify/1361432878835044352).

You'll need to rebase to pull in the `go.sum` update in https://github.com/kubernetes/release/pull/1918.

**Comment by saschagrunert** [comment]
Rebased on top of the latest master to fix the CI issues.

**Pull request merged**